components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -3174,11 +3199,14 @@ public void deleteIdPByResourceId(String resourceId, int tenantId, String tenant
                 throw new IdentityProviderManagementException(String.format(msg, resourceId, tenantDomain));
             }
             deleteIdP(dbConnection, tenantId, null, resourceId);
+            IdpMgtServiceComponentHolder.getInstance().getIdpSecretsProcessorService().deleteIdpSecrets(identityProvider);
```

Comment:
Add a comment above the delete secrets method

Reply:
Resolved with [039da67](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/039da6750bc84e5e950df73f8c057a0cf6567a87)